melbourne-school-of-theology.csl

```diff
@@ -496,24 +574,32 @@
       </if>
       <else-if variable="publisher-place publisher" match="any">
         <group delimiter="; " prefix=" (" suffix=")">
+          <group delimiter="; ">
+            <text macro="container-contributors-bibliography"/>
+            <text macro="locators-note"/>
+            <text macro="collection-title"/>
+          </group>
           <group>
-            <text term="volume" form="short" suffix=" "/>
-            <number variable="volume"/>
             <choose>
-              <if match="any" variable="number-of-volumes">
-                <text value="of" prefix=" " suffix=" "/>
+              <if match="all" variable="volume">
+                <text term="volume" form="short" suffix=" "/>
+                <number variable="volume"/>
+                <choose>
+                  <if match="any" variable="number-of-volumes">
+                    <text value="of" prefix=" " suffix=" "/>
+                  </if>
+                </choose>
               </if>
             </choose>
             <group>
               <number variable="number-of-volumes" form="numeric"/>
-              <text term="volume" form="short" plural="true" strip-periods="true" prefix=" "/>
+              <choose>
+                <if match="none" variable="volume">
```

Comment:
this can't be right - you only write "vols" if there is no volume number???. What you had before was probably right. If you have a term and a variable in a group, the term only gets printed if the variable is present.